src/config.ts

```diff
@@ -86,6 +86,7 @@ async function fetchProvider(source: ProviderSource): Promise<Provider | undefin
 }
 
 async function fetchAccount(source: AccountSource, provider: Provider): Promise<string | undefined> {
+  // XXX if we are only returning the address how do we sign the transaction?
```

Comment:
Signatures are handled by the provider in the case of unlocked accounts. Hm, for private-key based accounts, you may be right that we need to handle it ourselves since I guess it's not unlocked, it's something we're adding. Good point.